core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java

```diff
@@ -118,6 +119,24 @@ public void testRightEquiJoin()
                                                 tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
     }
 
+    @Test
+    public void testRightEquiJoinWIthLeftExpression()
+    {
+        assertPlan("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey",
+                output(
+                        join(
+                                RIGHT,
+                                ImmutableList.of(equiJoinClause("ORDERS_OK", "expr")),
+                                ImmutableMap.of("ORDERS_OK", "expr"),
```

Comment:
that dynamic filter should be removed by `RemoveUnsupportedDynamicFilters`. Why it wasn't removed?

Reply:
why should it be removed?
dynamic filter looks like this:` "orderkey" = #df_260` so the expression is a symbol reference

full  plan:
```
Output[orderkey]
│   Layout: [orderkey:bigint]
└─ RightJoin[("orderkey" = "expr")][$hashvalue, $hashvalue_3]
   │   Layout: [orderkey:bigint]
   │   dynamicFilterAssignments = {expr -> #df_260}
   ├─ ScanFilterProject[table = local:orders:sf0.01, filterPredicate = true, dynamicFilters = {"orderkey" = #df_260}]
   │      Layout: [orderkey:bigint, $hashvalue:bigint]
   │      $hashvalue := combine_hash(bigint '0', COALESCE("$operator$hash_code"("orderkey"), 0))
   │      orderkey := tpch:orderkey
   │      tpch:orderstatus
   │          :: [[F], [O], [P]]
   └─ LocalExchange[SINGLE] ()
      │   Layout: [expr:bigint, $hashvalue_3:bigint]
      └─ Project[]
         │   Layout: [expr:bigint, $hashvalue_4:bigint]
         │   $hashvalue_4 := combine_hash(bigint '0', COALESCE("$operator$hash_code"("expr"), 0))
         └─ ScanProject[table = local:lineitem:sf0.01]
                Layout: [expr:bigint]
                expr := ("orderkey_1" + BIGINT '1')
                orderkey_1 := tpch:orderkey

```

Reply:
see `io.trino.sql.planner.iterative.rule.RemoveUnsupportedDynamicFilters.Rewriter#isSupportedDynamicFilterExpression`
On probe side, we only support `CAST` and `symbol references`. Hence `LINEITEM_OK + BIGINT '1'` is not supported

Reply:
If I understand the above plan correctly, then the `lineitem` scan is built, and the `orders` scan is a probe side.
Since `LINEITEM_OK + BIGINT '1' `is pushed to the scan by PredicatePushDown the dynamic filter becomes :
`ORDERS_OK = expr` where ORDERS_OK is probe side and it's a symbol reference (I guess it doesn't matter at this point as `expr` is also a symbol reference).

Reply:
yes, build side complex expressions are fine